### PR TITLE
Create SearchBar and filters

### PR DESCRIPTION
**Pull request by agustingodoyc**
**STORY**
Create SearchBar and filters
**DESCRIPTION**
Create SearchBar and filters (PickerView)
**EVIDENCE**
Sep-01-2022 17-58-34

### CODE REVIEW

File: Ted Talk/Ted TalkTests/Ted_TalkTests.swift [hunk 1]
```diff
@@ -58,4 +60,23 @@ class Ted_TalkTests: XCTestCase {
...
+            talks = result
+            promise.fulfill()
+        }
+        
+        waitForExpectations(timeout: 5, handler: nil)
+        
+        talks = sut.serchWord(searchText: "x", picker: "description")
+        
+        XCTAssertEqual(talks?.count, 0, "Score computed from guess is wrong")
+    }
```
Comment (ezequielvelezglobant) [inline Ted Talk/Ted TalkTests/Ted_TalkTests.swift]: Separate into deferents test files, one for parser and other for dataManager

**Review by ezequielvelezglobant — CHANGES_REQUESTED** [review]
Please review the UI, the tableView should start right before pickerView

File: Ted Talk/Ted TalkTests/Ted_TalkTests.swift (hunk 1 above)
Comment (agustingodoyc) [reply Ted Talk/Ted TalkTests/Ted_TalkTests.swift]: [305e3cf](https://github.com/agustingodoyc/iOS_Academy_base_project/pull/3/commits/305e3cf6db927d984a3ad54c50acfef1563a4800)

File: Ted Talk/Ted Talk/Controller/ViewController.swift [hunk 2]
```diff
@@ -38,7 +76,40 @@ class ViewController: UIViewController, UITableViewDataSource {
...
+
+//MARK: - Picker View
+
+extension ViewController: UIPickerViewDelegate, UIPickerViewDataSource {
+    func numberOfComponents(in pickerView: UIPickerView) -> Int {
+        return 1
+    }
+    
+    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
+        return pickerOptions.allCases.count
```
Comment (JMVarelaGlobant) [inline Ted Talk/Ted Talk/Controller/ViewController.swift]: Use **filters** computed property here to be consistent all around the class

File: Ted Talk/Ted Talk/Controller/ViewController.swift [hunk 3]
```diff
@@ -38,7 +76,40 @@ class ViewController: UIViewController, UITableViewDataSource {
...
+    }
+    
+    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
+        return pickerOptions.allCases[row].rawValue
+    }
```
Comment (JMVarelaGlobant) [inline Ted Talk/Ted Talk/Controller/ViewController.swift:98]: Same here

**Review by JMVarelaGlobant — CHANGES_REQUESTED** [review]

File: Ted Talk/Ted Talk/Controller/ViewController.swift (hunk 3 above)
Comment (agustingodoyc) [reply Ted Talk/Ted Talk/Controller/ViewController.swift:98]: [Fixed Picker](https://github.com/agustingodoyc/iOS_Academy_base_project/pull/3/commits/03aad10796cbbe518724ad3fa2cadaa40e6cbc0e)

File: Ted Talk/Ted Talk/Controller/ViewController.swift (hunk 2 above)
Comment (agustingodoyc) [reply Ted Talk/Ted Talk/Controller/ViewController.swift]: [Fixed Picker](https://github.com/agustingodoyc/iOS_Academy_base_project/pull/3/commits/03aad10796cbbe518724ad3fa2cadaa40e6cbc0e)

**Review by JMVarelaGlobant — APPROVED** [review]

**Review by ezequielvelezglobant — APPROVED** [review]

**Pull request merged**